src/Manifest.cc

```diff
@@ -415,20 +415,28 @@ validateDepName(const StringRef name) {
     throw PoacError("dependency name must start with an alphanumeric character"
     );
   }
-  // Trailing `-`, `_`, and `/` are not allowed.
+  // Trailing `-`, `_`, `/`, and `.` are not allowed.
   if (!std::isalnum(name.back())) {
     throw PoacError("dependency name must end with an alphanumeric character");
   }
 
-  // Only alphanumeric characters, `-`, `_`, and `/` are allowed.
+  // Only alphanumeric characters, `-`, `_`, `/`, and `.` are allowed.
   for (const char c : name) {
-    if (!std::isalnum(c) && c != '-' && c != '_' && c != '/') {
-      throw PoacError("dependency name must be alphanumeric, `-`, `_`, or `/`");
+    if (!std::isalnum(c) && c != '-' && c != '_' && c != '/' && c != '.') {
+      throw PoacError(
+          "dependency name must be alphanumeric, `-`, `_`, `/`, or `.`"
+      );
     }
   }
 
-  // Consecutive `-`, `_`, and `/` are not allowed.
-  for (usize i = 1; i < name.size(); ++i) {
+  // `.` must be surrounded by numbers.
+  for (usize i = 1; i < name.size() - 1; ++i) {
+    if (name[i] == '.'
+        && (!std::isdigit(name[i - 1]) || !std::isdigit(name[i + 1]))) {
+      throw PoacError("dependency name must have a `.` surrounded by numbers");
+    }
+
+    // Consecutive `-`, `_`, `/`, and `.` are not allowed.
```

Comment:
Now, this is incorrect.  Previously, we were iterating over `1..name.size()` but now `1..name.size() - 1`.